src/mito2/src/flush.rs

```diff
@@ -62,23 +63,91 @@ pub trait WriteBufferManager: Send + Sync + std::fmt::Debug {
 
 pub type WriteBufferManagerRef = Arc<dyn WriteBufferManager>;
 
-// TODO(yingwen): Implements the manager.
+/// Default [WriteBufferManager] implementation.
+///
+/// Inspired by RocksDB's WriteBufferManager.
+/// <https://github.com/facebook/rocksdb/blob/main/include/rocksdb/write_buffer_manager.h>
 #[derive(Debug)]
-pub struct WriteBufferManagerImpl {}
+pub struct WriteBufferManagerImpl {
+    /// Write buffer size for the engine.
+    global_write_buffer_size: usize,
+    /// Mutable memtable memory size limitation.
+    mutable_limitation: usize,
```

Comment:
[What are the differences between limit and limitation?](https://www.quora.com/What-are-the-differences-between-limit-and-limitation#:~:text=A%20limit%20is%20something%20that,limit%20is%20a%20measurable%20quantity.)
```suggestion
    mutable_limit: usize,
```